#ifndef GUI_DECLINE_STORE_ITEM_DLG_H
#define GUI_DECLINE_STORE_ITEM_DLG_H

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Includes
//
#ifndef FM_CORE_H
#	include "fm_core.h"
#endif

// Ui
#include "ui_declineproduct.h"

// Qt Includes
#include <QDialog>
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
namespace fm {
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
namespace gui {
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// class CDeclineStoreItem
//
class CDeclineStoreItem : public QDialog
{
	Q_OBJECT

public:// Constructors
	CDeclineStoreItem( int nAvailableCount,
				   QWidget* pwParent = nullptr );
	~CDeclineStoreItem() = default;

public:
	//
	//	Main Interface
	//
	// Clear Dialog
	inline void Clear();
	inline void SetAvailableCount( int nCount );
	inline int  GetDeclineCount() const;

protected slots:
    void OnCountChanged( int nNewCount );
    //void OnSubtractClicked();

protected:// Helper Methodes
	void SetLabelColor( QLabel* pLabel, bool bValid );

private:
	//
	//	Content
	//
	Ui::declineStoreItemDlg ui;
	int m_nAvailableCount;
};
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Inline Implementations
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

// Clear
inline void CDeclineStoreItem::Clear()
{
	m_nAvailableCount = 0;
	ui.sbxSubtractCount->setValue( 0 );
}

// SetAvailableCount
inline void CDeclineStoreItem::SetAvailableCount( int nCount )
{
	if (nCount > 0)
	{
		m_nAvailableCount = nCount;
		ui.sbxSubtractCount->setMaximum( m_nAvailableCount );
	}
}

// SetAvailableCount
inline int CDeclineStoreItem::GetDeclineCount() const
{
	FM_ASSERT( ui.sbxSubtractCount );
	return ui.sbxSubtractCount->value();
}

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
} // namespace gui
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
} // namespace fm
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
#endif // GUI_DECLINE_STORE_ITEM_DLG_H